src/model/api.js

```diff
@@ -28,28 +27,33 @@ wsConnection.onopen = (event) => {
 }
 
 function getVersion() {
-  API.get("/api/version").then(res => {
-    console.log("Backend version: ", res.data)
-    version = versionToNumber(res.data)
-    console.log("Using numerical version: ", version)
-  })
-  .catch(error => {
-    console.log("Error getting version, using 0")
-    version = 0
+  return new Promise((resolve, reject) => {
+    API.get("/api/version")
+      .then(res => {
+        console.log("Backend version: ", res.data)
+        version = versionToNumber(res.data)
+        console.log("Using numerical version: ", version)
+        resolve(version)
+      })
+      .catch(error => {
+        console.log("Error getting version, using 0")
+        version = 0
+        resolve(version)
+      })
   })
 }
 
 function versionToNumber(versionStr) {
```

Comment:
You can actually compare version strings without converting them to numbers. `"4.1.11" < "4.1.12"` returns true